.github/workflows/main.yml

```diff
@@ -17,53 +17,76 @@ jobs:
   decision:
     name: Decision
     runs-on: ubuntu-20.04
+    outputs:
+      skipped: ${{ steps.skipDecision.outputs.result }}
+      platforms: ${{ steps.platformDecision.outputs.result }}
     steps:
-      # If an identical build exists that suceeded, skip this workflow run. We don't do
-      # this check for pull_request and merge_group events, out of caution.
-      - name: Skip previous identical builds
-        if: ${{ github.event_name != 'pull_request' && github.event_name != 'merge_group' }}
+      - name: Skip Decision
+        id: skipDecision
         uses: actions/github-script@v6
         with:
+          result-encoding: string
           script: |
+            // Never skip workflow runs for pull requests or merge groups, which might
+            // need to actually run / retry WPT tests.
+            if (context.eventName == "pull_request" || github.event_name == "merge_group") {
+              return "run";
+            }
+            // Skip the run if an identical run already exists. This helps to avoid running
+            // the workflow over and over again for the same commit hash.
             if ((await github.rest.actions.listWorkflowRuns({
               owner: context.repo.owner,
               repo: context.repo.repo,
               workflow_id: "main.yml",
               head_sha: context.sha,
               status: "success",
-            })).data.workflow_runs.length > 0)
-              await github.rest.actions.cancelWorkflowRun({
-                owner: context.repo.owner,
-                repo: context.repo.repo,
-                run_id: context.runId,
-              });
+            })).data.workflow_runs.length > 0) {
+              return "skip"
+            } else {
+              return "run"
+            }
+      - name: Platform Decision
+        id: platformDecision
+        uses: actions/github-script@v6
+        with:
+          result-encoding: string
+          script: |
+            if ("${{ steps.skipDecision.outputs.result }}" == "skip") {
+              return "none";
+            }
+            if (context.eventName == "push" || context.eventName == "merge_group") {
```

Comment:
I'm not able to find documentation for context.eventName? Is it an alias for 'github.event_name'?

Reply:
This comes from the [github-script documentation(https://github.com/actions/github-script#actionsgithub-script). I think it has a different way of working than GitHub Action code that's outside github-script actions.

Reply:
Thanks for the link!